sympy/solvers/diophantine.py

```diff
@@ -182,7 +182,7 @@ def diophantine(eq, param=symbols("t", integer=True), syms=None,
             if syms != var:
                 dict_sym_index = dict(zip(syms, range(len(syms))))
                 return {tuple([t[dict_sym_index[i]] for i in var])
-                            for t in diophantine(eq, param)}
+                            for t in diophantine(eq, param, permute = permute)}
```

Comment:
It's unusual to have spaces around `=` in function parameters. (PEP8)

Reply:
I have removed the spaces around =. And added a test.